### DISCUSSION TITLE
Using vitest for testing server-side Node.js applications

### DISCUSSION THREAD

**Discussion by i9or** [vitest-dev/vitest#1266] [General]
Hi everyone,

I have started evaluating vitest for my company's React.js codebases. And really love it so far! However, we also have quite a few Node.js applications. And having to support two test runners (Jest for servers apps) is not ideal. I tried to search for any examples of using vitest on the internet. Found a few, which is fine knowing that vitest is not yet very popular. I see that configuration is a bit different, as server apps do not use vite. So, my question is, are there any recommendations for or against using vitest for testing server apps? Or I will soon find some limitations? As clearly vitest is meant to be used in conjunction with vite. Thanks in advance for the answer.

**Comment by sheremet-va** [comment]
Vite is used to transform files, just like jest uses babel to transform files. So I wouldn't say that it _meant_ to be used with Vite applications.

By default Vitest doesn't come with `jsdom`, and tries to be lightweight. I've seen people use Vitest to test their server applications, so I don't see what can stop you from doing so. 

Technically, you should be able to run your jest tests with Vitest just by installing Vitest and runnin `vitest` command, since their API is largely the same, and all the differences are described in [Migration Guide](https://vitest.dev/guide/migration.html).

**Reply by i9or** [reply]
Thanks a lot for such a quick answer!

**Reply by Alino** [reply]
I have migrated our server side nodejs micro-service API from jest to vitest and it's working fine and blazingly fast :) feels better than jest.
